vector/sdks/rag-chat/quickstarts/cloudflare-workers.mdx

```diff
@@ -0,0 +1,100 @@
+---
+title: Cloudflare Workers
+---
+<Card title="GitHub Repository" icon="github" href="https://github.com/upstash/rag-chat/tree/master/examples/cloudflare-workers" horizontal>
+  You can find the project source code on GitHub.
+</Card>
+This guide provides detailed, step-by-step instructions on how to use Upstash RAG Chat with Cloudflare Workers. You can also explore our [Cloudflare Workers - Hono example](https://github.com/upstash/rag-chat/tree/master/examples/nextjs/cloudflare-workers) for detailed, end-to-end examples and best practices.
+
+### Project Setup
+
+Create a new Cloudflare Worker and install `@upstash/rag-chat` package.
+
+```shell
+npm create cloudflare@latest -- my-first-worker
+cd my-first-worker
+npm install @upstash/rag-chat
+```
+
+### Setup Upstash Redis
+
+Create a Redis database using [Upstash Console](https://console.upstash.com) or [Upstash CLI](https://github.com/upstash/cli) and copy the `UPSTASH_REDIS_REST_URL` and `UPSTASH_REDIS_REST_TOKEN` into your `.dev.vars` file.
+
+```shell .dev.vars
+UPSTASH_REDIS_REST_URL=<YOUR_URL>
+UPSTASH_REDIS_REST_TOKEN=<YOUR_TOKEN>
+```
+
+### Setup Upstash Vector
+
+Create a Vector index using [Upstash Console](https://console.upstash.com) or [Upstash CLI](https://github.com/upstash/cli) and copy the `UPSTASH_VECTOR_REST_URL` and `UPSTASH_VECTOR_REST_TOKEN` into your `.dev.vars` file.
+
+```shell .dev.vars
+UPSTASH_VECTOR_REST_URL=<YOUR_URL>
+UPSTASH_VECTOR_REST_TOKEN=<YOUR_TOKEN>
+```
+
+### Setup QStash LLM
+
+Navigate to [QStash Console](https://console.upstash.com/qstash) and copy the `QSTASH_TOKEN` into your `.dev.vars` file.
+
+```shell .dev.vars
+QSTASH_TOKEN=<YOUR_TOKEN>
+```
+
+### Setup Cloudflare Worker
+
+Update `/src/index.ts`:
+
+```typescript /src/index.ts
+import { RAGChat, upstash } from "@upstash/rag-chat";
+import { Index } from "@upstash/vector";
+import { Redis } from "@upstash/redis/cloudflare";
+
+export interface Env {
+	UPSTASH_REDIS_REST_TOKEN: string;
+	UPSTASH_REDIS_REST_URL: string;
+	UPSTASH_VECTOR_REST_TOKEN: string;
+	UPSTASH_VECTOR_REST_URL: string;
+	QSTASH_TOKEN: string;
+  }
+
+export default {
+	async fetch(request, env, ctx): Promise<Response> {
+		const ragChat = new RAGChat({
+			redis: new Redis({ token: env.UPSTASH_REDIS_REST_TOKEN, url: env.UPSTASH_REDIS_REST_URL }),
+			vector: new Index({ token: env.UPSTASH_VECTOR_REST_TOKEN, url: env.UPSTASH_VECTOR_REST_URL, cache: false }),
```

Comment:
can we replace the tab chars with space and put the token/url/cache to new lines?

We should probably highlight the cache param too